vortex-array/src/vtable/compute.rs

```diff
@@ -0,0 +1,30 @@
+use vortex_error::VortexResult;
+
+use crate::compute::{ComputeFn, InvocationArgs, Output};
+use crate::vtable::VTable;
+
+pub trait ComputeVTable<V: VTable> {
+    /// Dynamically invokes the given compute function on the array.
+    ///
+    /// Returns `None` if the compute function is not supported by this array, otherwise attempts
+    /// to invoke the function.
+    ///
+    /// This can be useful to support compute functions based on some property of the function,
+    /// without knowing at compile-time what that function is. For example, any elementwise
+    /// function can be pushed-down over a dictionary array and evaluated only on the unique values.
+    fn invoke(
+        array: &V::Array,
+        compute_fn: &ComputeFn,
+        args: &InvocationArgs,
+    ) -> VortexResult<Option<Output>>;
+}
+
+impl<V: VTable> ComputeVTable<V> for () {
```

Comment:
what is going on here?

Reply:
ok i see the comment in top-level VTable

Reply:
In theory, everything could be impl'd for () because the VTable type is in the generic.

It's just by convention that I chose to "impl ComputeVTable<ConstantVTable> for ConstantVTable" and use () to represent some default thing.

Maybe a better thing would be create an empty struct called "NotSupported". That would be a lot easier to read.